### PR TITLE
Refactor signature verification without traits

### PR DESCRIPTION
**Pull request by mystenmark**

### CODE REVIEW

File: crates/sui-types/src/message_envelope.rs [hunk 1]
```diff
@@ -131,19 +122,22 @@ impl<T: Message> Envelope<T, EmptySignInfo> {
     }
 }
 
-impl<T: Message + AuthenticatedMessage> Envelope<T, EmptySignInfo> {
-    pub fn verify_signature(&self, verify_params: &VerifyParams) -> SuiResult {
-        self.data.verify_message_signature(verify_params)
+impl Transaction {
```
Comment (lxfind) [inline crates/sui-types/src/message_envelope.rs:125]: This should be in `transaction.rs`

File: crates/sui-types/src/message_envelope.rs [hunk 2]
```diff
@@ -177,62 +171,59 @@ where
...
-        ))
+        Ok(VerifiedSignedTransaction::new_from_verified(self))
     }
 }
 
-impl<T> Envelope<T, AuthoritySignInfo>
-where
-    T: Message + UnauthenticatedMessage + Serialize,
-{
+impl SignedCheckpointSummary {
```
Comment (lxfind) [inline crates/sui-types/src/message_envelope.rs:211]: This should be in messages_checkpoint.rs

Comment (mystenmark) [reply crates/sui-types/src/message_envelope.rs:211]: yeah, meant to move these, thanks

File: crates/sui-types/src/message_envelope.rs [hunk 3]
```diff
@@ -281,58 +201,18 @@ where
...
-        self.auth_signature
-            .verify_secure(self.data(), Intent::sui_app(T::SCOPE), committee)
-    }
-}
-
-impl<T, const S: bool> Envelope<T, AuthorityQuorumSignInfo<S>>
-where
-    T: Message + UnauthenticatedMessage + Serialize,
-{
+impl CertifiedTransactionEffects {
```
Comment (lxfind) [inline crates/sui-types/src/message_envelope.rs:204]: This should be in effects

File: crates/sui-types/src/transaction.rs [hunk 4]
```diff
@@ -2259,17 +2273,6 @@ impl SenderSignedData {
 
         Ok(())
     }
-
-    pub fn verify_max_epoch_for_all_sigs(
```
Comment (lxfind) [inline crates/sui-types/src/transaction.rs:2263]: Could you point out why it is ok to remove this function?

File: crates/sui-types/src/transaction.rs [hunk 5]
```diff
@@ -2628,6 +2569,54 @@ pub type TrustedTransaction = TrustedEnvelope<SenderSignedData, EmptySignInfo>;
...
+        current_epoch: EpochId,
+        verify_params: &VerifyParams,
+    ) -> SuiResult<VerifiedTransaction> {
+        self.verify_signature(current_epoch, verify_params)?;
+        Ok(VerifiedTransaction::new_from_verified(self))
+    }
+}
+
+impl SignedTransaction {
+    pub fn verify_signatures_authenticated(
```
Comment (lxfind) [inline crates/sui-types/src/transaction.rs:2592]: Probably need better naming to distinguish these two.
Maybe add `_into` to the consuming version

File: crates/sui-types/src/transaction.rs (hunk 4 above)
Comment (mystenmark) [reply crates/sui-types/src/transaction.rs:2263]: all authentication goes through this path now: https://github.com/MystenLabs/sui/blob/b89eb622bbbf9d1c64d6b2a8855548a6bc2f78f2/crates/sui-types/src/signature.rs#L109

File: crates/sui-types/src/transaction.rs (hunk 5 above)
Comment (mystenmark) [reply crates/sui-types/src/transaction.rs:2592]: Done, made it `try_into_verified` since its faillable

**Review by lxfind — APPROVED** [review]
Thanks!

**Review by joyqvq — APPROVED** [review]

**Pull request merged**